pre2p/types/validator.go

```diff
@@ -0,0 +1,40 @@
+package types
+
+type NodeId uint32
+
+type ValMap map[NodeId]*Validator
+
+// TODO(olshansky): Review all the attributes of this struct.
+type Validator struct {
+	NodeId     NodeId   `json:"node_id"`
```

Comment:
So many todo's haha. 

Similar to above, I'd recommend deprecating nodeId for the public key if possible. I see a TODO to remove it above, so won't harp or block cause of this

Reply:
Removing NodeId altogether. It'll make the consensus PR a little more complicated but I think it's the right move this early on.

Reply:
Note that in order to remove NodeId from everywhere, I had to change the Address type from a byte slice to a byte array (i.e. fixed length with size `AddressLen`). Lmk if that's an issue.

Reply:
Actually this is a problem

Protobuf only allows []byte not fixed length

So it's either we convert all slices to byte arrays using `copy` or we change address encoding to strings

Reply:
Discussed this offline with Andrew directly. The decision was:
- Leave `Address` as a slice rather than a string or array for proto compatibility and confirmation to standards
- In the map (where a key cannot be a slice), simply use a string (hex representation of the address) without needing to create an extra type.